Dockerfiles/agent/windows/amd64/Dockerfile

```
@@ -1,16 +1,24 @@
 ARG BASE_IMAGE=mcr.microsoft.com/powershell:lts-windowsservercore-1809
-FROM ${BASE_IMAGE}
-
 ARG WITH_JMX="false"
 ARG VARIANT="unknown"
 
+FROM mcr.microsoft.com/powershell:lts-windowsservercore-${VARIANT} as unzipper
+USER ContainerAdministrator
+SHELL ["pwsh", "-Command", "$ErrorActionPreference = 'Stop';"]
+COPY datadog-agent-latest.amd64.zip uncompress-zip.ps1 ./
+RUN . ./uncompress-zip.ps1
+
+FROM ${BASE_IMAGE}
+
 LABEL maintainer "Datadog <package@datadoghq.com>"
 
 USER ContainerAdministrator
 
 SHELL ["pwsh", "-Command", "$ErrorActionPreference = 'Stop';"]
 
-COPY datadog-agent-latest.amd64.zip install.ps1 ./
+COPY --from=unzipper ["C:/ProgramData/Datadog", "C:/ProgramData/Datadog"]
+COPY --from=unzipper ["C:/Program Files/Datadog", "C:/Program Files/Datadog"]
+COPY install.ps1 ./

```

For caching purpose it can be better to copy first `install.ps1`

```suggestion
COPY install.ps1 ./
COPY --from=unzipper ["C:/ProgramData/Datadog", "C:/ProgramData/Datadog"]
COPY --from=unzipper ["C:/Program Files/Datadog", "C:/Program Files/Datadog"]

```